staging/src/k8s.io/component-helpers/storage/volume/helpers.go

```diff
@@ -68,3 +69,20 @@ func CheckNodeAffinity(pv *v1.PersistentVolume, nodeLabels map[string]string) er
 
 	return nil
 }
+
+func CheckCSINodeDriverAffinity(csiNode *storageV1.CSINode, pv *v1.PersistentVolume) error {
+	if pv.Spec.CSI == nil {
+		return nil
+	}
+
+	err := fmt.Errorf("no matching CSI node driver")
+	if csiNode == nil || len(csiNode.Spec.Drivers) == 0 {
+		return fmt.Errorf("%s, csinode is empty or csinode drivers is empty", err.Error())
```

Comment:
```suggestion
	if csiNode == nil {
		return fmt.Errorf("CSINode not exist")
```
Keep the message concise. As the message may be shown in the events, possibly aggregated with many other messages reported for other nodes.

Reply:
The suggested error messages changes are reasonable, but we need to retain the condition check. If the CSI agent crashes or unserviceable, the driver list could be empty. We need to handle this scenario properly.

Reply:
I've reconfirmed that all bound volume nodes scheduler errors report a single event: 'node(s) had volume node affinity conflict.' These errors remain hidden from users. It would be beneficial to offer more specific details.

Reply:
> the driver list could be empty

This case can be covered by bellowing "no matching CSI driver registered" branch.